File: md5-authentication.py
#!/usr/bin/python
import sys
import hashlib
import time

try:
    from urllib.parse import urlsplit
except ImportError:
    from urlparse import urlsplit


def sign_url(url, password, expires=None):
    if expires is None:
        expires = int(time.time()) + (60 * 60 * 12)
    parts = urlsplit(url)
    if parts.query:
        sign = "%s?%s&expires=%s&pass=%s" % (
            parts.path, parts.query, expires, password)
        new = "%s://%s%s?%s&expires=%s&token=" % (
            parts.scheme, parts.netloc, parts.path, parts.query, expires)
    else:
        sign = "%s?expires=%s&pass=%s" % (parts.path, expires, password)
        new = "%s://%s%s?expires=%s&token=" % (
            parts.scheme, parts.netloc, parts.path, expires)
    md5instance = hashlib.md5()
    md5instance.update(sign.encode("ascii"))
    token = md5instance.hexdigest()
    return "%s%s" % (new, token)


expires = int(time.time()) + (60 * 60 * 12)
password = 'xa5aileeph6nah5ooQu'

for url in sys.argv[1:]:
    print sign_url(url, password, expires)

File: md5-authentication.php
#!/usr/bin/php
<?php
function sign_url($url, $password, $expires = false) {
    if ($expires === false) {
        $expires = time() + (60*60*12);
    }
    $parts = parse_url($url);
    if (isset($parts['query'])) {
        $sign = sprintf("%s?%s&expires=%s&pass=%s",
            $parts['path'], $parts['query'], $expires, $password);
        $new = sprintf("%s://%s%s?%s&expires=%s&token=",
            $parts['scheme'], $parts['host'], $parts['path'],
            $parts['query'], $expires);
    } else {
        $sign = sprintf("%s?expires=%s&pass=%s",
            $parts['path'], $expires, $password);
        $new = sprintf("%s://%s%s?expires=%s&token=",
            $parts['scheme'], $parts['host'], $parts['path'],
            $expires);
    }
    $token = md5($sign);
    return sprintf("%s%s", $new, $token);
}
$expires = time() + (60*60*12);
$password = "xa5aileeph6nah5ooQu";
foreach(array_slice($_SERVER['argv'], 1) as $url) {
    printf("%s\n", sign_url($url, $password, $expires));
}
